BidMachineSample/Native.m

```diff
@@ -0,0 +1,230 @@
+     ]];
+}
+
+- (void)layoutBidMachineNativeView {
+    NativeAdView *adView = [NativeAdView new];
+    NSError *error;
+    self.bidMachineNativeAd.controller = self;
+    [self.bidMachineNativeAd presentAd:self.adContainer :adView error:&error];
+    
+    if (error) {
+        return;
+    }
+    
+    adView.translatesAutoresizingMaskIntoConstraints = false;
+    [self.adContainer addSubview:adView];
+    [NSLayoutConstraint activateConstraints:
+     @[
+        [adView.topAnchor constraintEqualToAnchor:self.adContainer.topAnchor],
+        [adView.bottomAnchor constraintEqualToAnchor:self.adContainer.bottomAnchor],
+        [adView.leadingAnchor constraintEqualToAnchor:self.adContainer.leadingAnchor],
+        [adView.trailingAnchor constraintEqualToAnchor:self.adContainer.trailingAnchor],
+    ]];
+}
+
+- (void)layoutAdManagerNativeView {
+    GADNativeAdView* googleAdView = [GADNativeAdView new];
+    googleAdView.nativeAd = self.googleNativeAd;
+
+    // Set up GADNativeAdView following the instructions at: https://developers.google.com/ad-manager/mobile-ads-sdk/ios/native/advanced
+}
+
+- (void)deleteLoadedAd {
+    [self.bidMachineNativeAd unregisterView];
+
+    self.bidMachineNativeAd = nil;
+    self.googleNativeAd = nil;
+    self.adLoader = nil;
+    [self.adContainer.subviews enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
+        [subview removeFromSuperview];
+    }];
+}
+
+- (void)loadNative {
+    [self deleteLoadedAd];
+
+    __weak typeof(self) weakSelf = self;
+    [BidMachineSdk.shared native:nil :^(BidMachineNative *native, NSError *error) {
+        if (error) {
+            [weakSelf switchState: BSStateIdle];
+            NSLog(@"Native ad request finished with error %@",error.localizedDescription);
+            return;
+        }
+        weakSelf.bidMachineNativeAd = native;
+        weakSelf.bidMachineNativeAd.controller = weakSelf;
+        weakSelf.bidMachineNativeAd.delegate = weakSelf;
+        [weakSelf.bidMachineNativeAd loadAd];
+    }];
+}
+
+- (void)loadAdManagerNativeWith:(id<BidMachineAdProtocol> _Nullable)ad {
+    GAMRequest *googleRequest = [GAMRequest request];
+
+    if (ad) {
+        NSString *price = [self.formatter stringFromNumber:@(ad.auctionInfo.price)];
+        googleRequest.customTargeting = @{ @"bm_pf" : price };
+    }
+
+    self.adLoader = [
+        [GADAdLoader alloc]
+        initWithAdUnitID:@UNIT_ID
+        rootViewController:self
+        adTypes:@[ GADAdLoaderAdTypeNative ]
+        options:nil
+    ];
+
+    self.adLoader.delegate = self;
+    [self.adLoader loadRequest:googleRequest];
+}
+
+- (void)onBidMachineWin {
+    [BidMachineSdk.shared notifyMediationWin:self.bidMachineNativeAd];
+    
+    if (self.bidMachineNativeAd.canShow) {
+        [self switchState:BSStateReady];
+    }
+}
+
+- (void)onBidMachineLoss {
+    [BidMachineSdk.shared notifyMediationLoss:@"unknown" ecpm:0.0 ad:self.bidMachineNativeAd];
```

Comment:
Let's use empty string `@""` instead of `@"unknown"`

Reply:
✅